c10/cuda/CUDADeviceAssertion.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <c10/cuda/CUDAException.h>
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+namespace cuda {
+
+#ifdef TORCH_USE_CUDA_DSA
+// Copy string from `src` to `dst`
+static __device__ void dstrcpy(char* const dst, const char* const src) {
+  int i = 0;
+  // Copy string from source to destination, ensuring that it
+  // isn't longer than `C10_CUDA_DSA_MAX_STR_LEN-1`
+  for (; src[i] != '\0' && i < C10_CUDA_DSA_MAX_STR_LEN - 1; i++) {
+    dst[i] = src[i];
+  }
+  // If the loop terminated before `i` meets the condition below, then
+  // `dst` ends with a null-terminator from `src`. Otherwise, the loop
+  // reached maximum iterations and we add our own null-terminator to
+  // `dst`
+  if (i == C10_CUDA_DSA_MAX_STR_LEN - 1) {
+    *dst = '\0';
+  }
```

Comment:
This sounds like a while loop to me, that copies up to C10_CUDA_DSA_MAX_STR_LEN-1 characters from src to dst until `\0` is met
```suggestion
  while ( *src != '\0' && i++ < C10_CUDA_DSA_MAX_STR_LEN - 1) {
    *dst++ = *src++;
  }
  *dst = '\0';
```

Reply:
I've updated the code to adopt this suggestion.